README.md

```diff
@@ -96,14 +97,24 @@ in mkDerivation rec {
 }
 ```
 
+## How does it work?
+
+**mvn2nix** relies on [maven-invoker](https://maven.apache.org/shared/maven-invoker/); which fires off
+Maven in a separate JVM process.
+
+Maven is executed with a temporary *ephemeral* local repository for the given goals provided (defaults to **package**).
+The local repository is than traversed, and each encountered file is recorded in the dependencies list.
```

Comment:
Does it use the existing `.m2` like a remote cache?
The answer has implications for performance and the behavior around SNAPSHOTs and other jarringly sneaky jar updates.

Reply:
By existing `.m2` do you mean the default local repository (~/.m2) ? 

The artifacts must be downloaded from the remote repository (i.e. maven-central) each time to calculate the SHA256; so I do not see a way to avoid that by using the default local repository.

I suspect; I should add a safeguard to simply fail if any _SNAPSHOT_ dependencies are found.
Of course the hermetic aspect of this is somewhat confusing if having any SNAPSHOT dependency.

Since maven should be run "offline"; whatever SNAPSHOT is fetched at the moment of generated dependencies.nix at the moment would be used.

Reply:
> The artifacts must be downloaded from the remote repository (i.e. maven-central) each time to calculate the SHA256; so I do not see a way to avoid that by using the default local repository.

This seems to work.

```console
$ curl -s https://repo.maven.apache.org/maven2/com/google/errorprone/error_prone_annotations/2.3.4/error_prone_annotations-2.3.4.jar | sha256sum -
baf7d6ea97ce606c53e11b6854ba5f2ce7ef5c24dddf0afa18d1260bd25b002c  -

$ sha256sum ~/.m2/repository/com/google/errorprone/error_prone_annotations/2.3.4/error_prone_annotations-2.3.4.jar
baf7d6ea97ce606c53e11b6854ba5f2ce7ef5c24dddf0afa18d1260bd25b002c  /home/user/.m2/repository/com/google/errorprone/error_prone_annotations/2.3.4/error_prone_annotations-2.3.4.jar

```

> add a safeguard to simply fail if any SNAPSHOT dependencies are found.

:100: 

Does `mvn2nix` support multi-module projects? That might interact with the safeguard, because SNAPSHOT dependencies are ok if built locally.

Reply:
@roberth the multi-modules though wouldn't be found in the local repository.

If you build a multi-module project; they are all included in the reactor together.
I will include a dummy example though :+1: 

I will create an issue to record the improvement to see if ~/.m2/repository can be set to be a remote repository for the ephemeral repository.
A "fresh" repository is needed though so i can determine the exact closure for only the current application under question.

Reply:
Please see https://github.com/fzakaria/mvn2nix/pull/12 for using ~/.m2 as a remote repository